commands/utils/mavenutils.go

```diff
@@ -13,46 +13,79 @@ func GetVersionProperties(projectPath string, depToPropertyMap map[string][]stri
 	if err != nil {
 		return err
 	}
-	dependenciesPattern := "(<dependency>(.|\\s)*?<\\/dependency>)|(<plugin>(.|\\s)*?<\\/plugin>)"
-	dependenciesRegexp, err := regexp.Compile(dependenciesPattern)
+	mavenDependencies, err := getDependenciesFromPomXml(contentBytes)
 	if err != nil {
 		return err
 	}
-	dependencyStrings := dependenciesRegexp.FindAll(contentBytes, -1)
-	for _, depStr := range dependencyStrings {
-		dep := &mavenDependency{}
-		err = xml.Unmarshal(depStr, dep)
-		if err != nil {
-			return err
-		}
-		depName := dep.GroupId + ":" + dep.ArtifactId
+	for _, mavenDependency := range mavenDependencies {
+		depName := mavenDependency.GroupId + ":" + mavenDependency.ArtifactId
 		if _, exist := depToPropertyMap[depName]; !exist {
 			depToPropertyMap[depName] = []string{}
 		}
-		if strings.HasPrefix(dep.Version, "${") {
-			depToPropertyMap[depName] = append(depToPropertyMap[dep.GroupId+":"+dep.ArtifactId], strings.TrimPrefix(strings.TrimSuffix(dep.Version, "}"), "${"))
+		if strings.HasPrefix(mavenDependency.Version, "${") {
+			depToPropertyMap[depName] = append(depToPropertyMap[mavenDependency.GroupId+":"+mavenDependency.ArtifactId], strings.TrimPrefix(strings.TrimSuffix(mavenDependency.Version, "}"), "${"))
 		}
 	}
 
-	modulePattern := "<module>(.|\\s)*?<\\/module>"
-	moduleRegexp, err := regexp.Compile(modulePattern)
+	moduleStrings, err := getMavenModuleFromPomXml(contentBytes)
 	if err != nil {
 		return err
 	}
-	moduleStrings := moduleRegexp.FindAllString(string(contentBytes), -1)
 	for _, moduleStr := range moduleStrings {
-		modulePath := strings.TrimPrefix(strings.TrimSuffix(moduleStr, "</module>"), "<module>")
-		modulePath = strings.TrimSpace(modulePath)
-		err = GetVersionProperties(filepath.Join(projectPath, modulePath), depToPropertyMap)
-		if err != nil {
+		if err = GetVersionProperties(filepath.Join(projectPath, moduleStr), depToPropertyMap); err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
+// Extract all dependencies from the input pom.xml
+// pomXmlContent - The pom.xml content
+func getDependenciesFromPomXml(pomXmlContent []byte) ([]mavenDependency, error) {
+	dependenciesPattern := "(<dependency>(\\S|\\s)*?<\\/dependency>)|(<plugin>(\\S|\\s)*?<\\/plugin>)"
+	dependenciesRegexp, err := regexp.Compile(dependenciesPattern)
```

Comment:
Since compiling regular expressions is expenssive, it's best to do it only once per expression.
The same goes for the `getMavenModuleFromPomXml` function.